java.lang.IllegalStateException: missing file for budgets : /var/run/zilla/budgets127

Zilla crashes with the below error

Steps to reproduce the behavior:
1. Run the [Zilla quickstart](https://docs.aklivity.io/zilla/latest/tutorials/quickstart/kafka-proxies.html)
2. wait 3-5 min
4. The below error happens and the server will restart
5. This repeats every 3-5 min

```
 org.agrona.concurrent.AgentTerminationException: java.lang.IllegalStateException: missing file for budgets : /var/run/zilla/budgets127
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:707)
     at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
     at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
     at java.base/java.lang.Thread.run(Thread.java:1623)
 Caused by: java.lang.IllegalStateException: missing file for budgets : /var/run/zilla/budgets127
     at org.agrona.core/org.agrona.IoUtil.checkFileExists(IoUtil.java:507)
     at org.agrona.core/org.agrona.IoUtil.mapExistingFile(IoUtil.java:396)
     at org.agrona.core/org.agrona.IoUtil.mapExistingFile(IoUtil.java:361)
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.layouts.BudgetsLayout$Builder.build(BudgetsLayout.java:144)
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.newBudgetDebitor(DispatchAgent.java:1620)
     at org.agrona.core/org.agrona.collections.Int2ObjectHashMap.computeIfAbsent(Int2ObjectHashMap.java:268)
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.supplyDebitor(DispatchAgent.java:536)
     at io.aklivity.zilla.runtime.binding.kafka@0.9.58/io.aklivity.zilla.runtime.binding.kafka.internal.budget.KafkaMergedBudgetAccountant.supplyDebitor(KafkaMergedBudgetAccountant.java:49)
     at io.aklivity.zilla.runtime.binding.kafka@0.9.58/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheClientFetchFactory$KafkaCacheClientFetchStream.onClientReplyWindow(KafkaCacheClientFetchFactory.java:1645)
     at io.aklivity.zilla.runtime.binding.kafka@0.9.58/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheClientFetchFactory$KafkaCacheClientFetchStream.onClientMessage(KafkaCacheClientFetchFactory.java:981)
     at io.aklivity.zilla.runtime.binding.kafka@0.9.58/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheClientFetchFactory.lambda$newStream$2(KafkaCacheClientFetchFactory.java:248)
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadReply(DispatchAgent.java:1276)
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:1045)
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
     at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:701)
     ... 3 more
     Suppressed: java.lang.Exception: [engine/data#0]        [0x00000000000001fe] streams=[consumeAt=0x00057e78 (0x0000000000057e78), produceAt=0x00058118 (0x0000000000058118)]
             at io.aklivity.zilla.runtime.engine@0.9.58/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:705)
             ... 3 more
```

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java
@@ -20,7 +20,7 @@ import static io.aklivity.zilla.runtime.binding.kafka.internal.types.KafkaCapabi
 import static io.aklivity.zilla.runtime.binding.kafka.internal.types.KafkaOffsetType.HISTORICAL;
 import static io.aklivity.zilla.runtime.binding.kafka.internal.types.KafkaOffsetType.LIVE;
 import static io.aklivity.zilla.runtime.binding.kafka.internal.types.stream.WindowFW.Builder.DEFAULT_MINIMUM;
-import static io.aklivity.zilla.runtime.engine.budget.BudgetCreditor.NO_CREDITOR_INDEX;
+import static io.aklivity.zilla.runtime.engine.budget.BudgetCreditor.NO_BUDGET_ID;
 
 import java.util.ArrayList;
 import java.util.Collections;
@@ -1044,7 +1044,7 @@ public final class KafkaMergedFactory implements BindingHandler
         private int nextNullKeyHashData;
         private int nextNullKeyHashFlush;
         private int fetchStreamIndex;
-        private long mergedReplyBudgetId = NO_CREDITOR_INDEX;
+        private long mergedReplyBudgetId = NO_BUDGET_ID;
 
         private KafkaUnmergedConsumerStream consumerStream;
         private KafkaUnmergedProduceStream producer;
@@ -1431,13 +1431,13 @@ public final class KafkaMergedFactory implements BindingHandler
             if (KafkaState.replyOpening(state))
             {
                 state = KafkaState.openedReply(state);
-                if (mergedReplyBudgetId == NO_CREDITOR_INDEX)
+                if (mergedReplyBudgetId == NO_BUDGET_ID)
                 {
                     mergedReplyBudgetId = creditor.acquire(replyId, budgetId);
                 }
             }
 
-            if (mergedReplyBudgetId != NO_CREDITOR_INDEX)
+            if (mergedReplyBudgetId != NO_BUDGET_ID)
             {
                 creditor.credit(traceId, mergedReplyBudgetId, credit);
             }
@@ -1510,7 +1510,7 @@ public final class KafkaMergedFactory implements BindingHandler
             if (!KafkaState.replyOpened(state) && replyBudget > 0)
             {
                 state = KafkaState.openedReply(state);
-                if (mergedReplyBudgetId == NO_CREDITOR_INDEX)
+                if (mergedReplyBudgetId == NO_BUDGET_ID)
                 {
                     mergedReplyBudgetId = creditor.acquire(replyId, replyBudgetId);
                     creditor.credit(traceId, mergedReplyBudgetId, replyBudget);
@@ -1791,10 +1791,10 @@ public final class KafkaMergedFactory implements BindingHandler
 
         private void cleanupBudgetCreditorIfNecessary()
         {
-            if (mergedReplyBudgetId != NO_CREDITOR_INDEX)
+            if (mergedReplyBudgetId != NO_BUDGET_ID)
             {
                 creditor.release(mergedReplyBudgetId);
-                mergedReplyBudgetId = NO_CREDITOR_INDEX;
+                mergedReplyBudgetId = NO_BUDGET_ID;
             }
         }
 

